engine/resources/user.go

```diff
@@ -333,6 +333,26 @@ func (obj *UserRes) Cmp(r engine.Res) error {
 type UserUID struct {
 	engine.BaseUID
 	name string
+	uid  *uint32
+}
+
+// IFF aka if and only if they are equivalent, return true. If not, false.
+func (obj *UserUID) IFF(uid engine.ResUID) bool {
+	res, ok := uid.(*UserUID)
+	if !ok {
+		return false
+	}
+	if obj.uid != nil && res.uid != nil {
```

Comment:
nit, but for consistency, the equivalent function in *GroupUID does name compare, and then the gui compare. Can you swap these two statements so they look otherwise closely equivalent please?

Reply:
Swapped

Reply:
GroupUID has gid check first, then name: https://github.com/purpleidea/mgmt/blob/master/engine/resources/group.go#L261
Want me to leave this, or swap both of them?